tv_apps/smart-home/js/edit.js

```diff
@@ -65,67 +63,75 @@
           .then(function() {
-            // Since writeCardlistInCardStore triggers card-removed event that
-            // causes re-focus on other elements in card list, we need to wait
-            // until those actions to be done before focusing editButton.
-            this.spatialNavigator.focus(this.editButton);
-            this.currentNode.classList.remove('focused');
+            this.spatialNavigator.focus(this.currentScrollable);
+            this.currentScrollable.realignToReferenceElement();
           }.bind(this));
         this._concealPanel(this.currentScrollable, this.currentNode);
         this.fire('exit-edit-mode');
       } else {
-        this.mainSection.dataset.mode = 'edit';
+        this.mode = 'edit';
         this.spatialNavigator.multiRemove(this.regularNavElements);
         this.spatialNavigator.multiAdd(this.editNavElements);
 
-        this._setCurrentScrollable(this.cardScrollable);
+        var scrollable = null;
+        if (this.spatialNavigator.getFocusedElement() ===
+          this.folderScrollable) {
+          scrollable = this.folderScrollable;
+        } else {
+          scrollable = this.cardScrollable;
+          this.folderScrollable.clean();
+        }
+
+        this._setCurrentScrollable(scrollable);
         this.currentNode =
-          this.cardScrollable.getNodeFromItem(this.cardScrollable.currentItem);
+          scrollable.getNodeFromItem(scrollable.currentItem);
         this._revealPanel(this.currentScrollable, this.currentNode);
 
-        this.spatialNavigator.focus(this.cardScrollable);
         this.cardScrollable.setScale(EDIT_MODE_SCALE);
         this.folderScrollable.setScale(EDIT_MODE_SCALE);
+        this.spatialNavigator.focus(scrollable);
+        scrollable.realignToReferenceElement();
         this.fire('enter-edit-mode');
       }
     },
 
     toggleArrangeMode: function() {
-      if (this.mainSection.dataset.mode === 'edit') {
-        this.mainSection.dataset.mode = 'arrange';
+      if (this.mode === 'edit') {
+        this.mode = 'arrange';
         this._concealPanel(this.currentScrollable, this.currentNode);
         this.currentScrollable.focus(this.currentScrollable.currentItem);
         this._setHintArrow();
         if (this.currentScrollable === this.folderScrollable) {
           this.cardScrollable.setColspanOnFocus(1);
         }
         this.fire('arrange');
-      } else if (this.mainSection.dataset.mode == 'arrange') {
-        this.mainSection.dataset.mode = 'edit';
+      } else if (this.mode == 'arrange') {
+        this.mode = 'edit';
 
         this.currentNode = this.currentScrollable.getNodeFromItem(
                                             this.currentScrollable.currentItem);
         this._revealPanel(this.currentScrollable, this.currentNode);
         this._clearHintArrow();
         this.cardScrollable.setColspanOnFocus(0);
         this.folderScrollable.realignToReferenceElement();
+        this.fire('arranged');
```

Comment:
Can be removed due to comments in https://github.com/mozilla-b2g/gaia/pull/34247/files#diff-5a33a94cd737c2a136a3c3139f952015R203.